store/store_download.go

```diff
@@ -318,6 +324,14 @@ func DownloadIcon(ctx context.Context, name string, targetPath string, downloadU
 		return err
 	}
 
+	// Read etag of existing file at targetPath, if it exists
+	var etag string
+	etagBuf := make([]byte, maxEtagSize)
+	if size, err := unix.Getxattr(targetPath, "user.snapstore-etag", etagBuf); err == nil {
```

Comment:
looking at conventions in the docs it seems this needs to be called user.snap_store_etag?

Reply:
also this string should be  a global const here I suppose